test/e2e/parallel/spacebindingrequest_test.go

```diff
@@ -0,0 +1,267 @@
+package parallel
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	"github.com/gofrs/uuid"
+
+	spacebindingrequesttestcommon "github.com/codeready-toolchain/toolchain-common/pkg/test/spacebindingrequest"
+
+	testspace "github.com/codeready-toolchain/toolchain-common/pkg/test/space"
+	. "github.com/codeready-toolchain/toolchain-e2e/testsupport"
+	. "github.com/codeready-toolchain/toolchain-e2e/testsupport/wait"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func TestCreateSpaceBindingRequest(t *testing.T) {
+	// given
+	t.Parallel()
+	// make sure everything is ready before running the actual tests
+	awaitilities := WaitForDeployments(t)
+	hostAwait := awaitilities.Host()
+	memberAwait := awaitilities.Member1()
+
+	t.Run("success", func(t *testing.T) {
+		t.Run("create space binding request", func(t *testing.T) {
+			// when
+			// we create a space to share , a new MUR and a SpaceBindingRequest
+			space, spaceBindingRequest, spaceBinding := NewSpaceBindingRequest(t, awaitilities, memberAwait, hostAwait, "admin")
+
+			t.Run("spaceBinding is recreated if deleted ", func(t *testing.T) {
+				// now, delete the SpaceBinding,
+				// a new SpaceBinding will be provisioned by the SpaceBindingRequest.
+				//
+				// save the creation timestamp that will be used to ensure that a new SpaceBinding was created with the same name.
+				oldSpaceCreationTimeStamp := spaceBinding.CreationTimestamp
+				// save the old UID that will be used to ensure that a new SpaceBinding was created with the same name but new UID
+				oldUID := spaceBinding.UID
+
+				// when
+				err := hostAwait.Client.Delete(context.TODO(), spaceBinding)
+
+				// then
+				// a new SpaceBinding is created
+				// with the same name but creation timestamp should be greater (more recent).
+				require.NoError(t, err)
+				spaceBinding, err = hostAwait.WithRetryOptions(TimeoutOption(time.Second*10), RetryInterval(time.Second*2)).WaitForSpaceBinding(t, spaceBindingRequest.Spec.MasterUserRecord, space.Name,
```

Comment:
I increased the retry interval and timeout because I'm noticing the test fails from time to time with this error :
```
    ----
        diffs:
        expected SpaceBinding to be created after 2023-08-17 08:43:00 +0000 UTC; Actual creation timestamp 2023-08-17 08:43:00 +0000 UTC
 ```
[ full logs here](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/codeready-toolchain_host-operator/848/pull-ci-codeready-toolchain-host-operator-master-e2e/1692082545362997248/build-log.txt)
I'm pretty sure the SpaceBinding get's deleted, because I've observed that and manually tested it on my temp cluster, but I think this `WaitForSpaceBinding` function is picking up the SpaceBinding before it get's deleted, for this reason I've tried waiting more and I've also added this  check on the UID : `UntilSpaceBindingHasDifferentUID(oldUID)`

maybe I can remove now the retry interval and see if it's more stable, or at least add a comment to explain why is there. WDYT?

Reply:
Ok, I've noticed after that the default timeout was actually bigger than what I was setting, so I've removed that and also removed the `creationTimeStamp` check ,I've left only the UID one, please see: https://github.com/codeready-toolchain/toolchain-e2e/pull/779/commits/88eaeb5080ea1d9a058b54dda7d45e98c1102305

Reply:
cool, thanks :+1: